core/common.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * @hidden
+ */
+export interface ICommonContext {
+  /**
+   * Returns a valid SQL string that can be used to reference the dataset produced by this action.
+   */
+  self: () => string;
+
+  /**
+   * Returns the name of this dataset.
+   */
+  name: () => string;
+
+  /**
+   * References another action, returning valid SQL to be used in a `from` expression and adds it as a dependency to this action.
```

Comment:
invert the order of clauses here?

`References another action, adding it as a dependency to this action, returning valid SQL to be used ...`